dff/pipeline/service/extra.py

```diff
@@ -168,6 +168,18 @@ def info_dict(self) -> dict:
 
 
 class BeforeHandler(_ComponentExtraHandler):
+    """
+    A handler for extra functions that are executed before the component's main function.
+
+    :param functions: A callable or a list of callables that will be executed
+        before the component's main function.
+    :type functions: ExtraHandlerBuilder
+    :param timeout: Optional timeout for the execution of the extra functions, in
+        seconds.
+    :param asynchronous: Optional flag that indicates whether the extra functions
+        should be executed asynchronously.
```

Comment:
Mention that the default value of the flag is True if all the functions in this handler are asynchronous.